microcloud/service/lxd_join.go

```diff
@@ -0,0 +1,75 @@
+package service
+
+import (
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/version"
+)
+
+// ErrCertificateExists indicates that a certificate already exists.
+var ErrCertificateExists error = fmt.Errorf("Certificate already in trust store")
+
+// SetupTrust is a convenience around InstanceServer.CreateCertificate that adds the given server certificate to
+// the trusted pool of the cluster at the given address, using the given password. The certificate is added as
+// type CertificateTypeServer to allow intra-member communication. If a certificate with the same fingerprint
+// already exists with a different name or type, then no error is returned.
+func SetupTrust(serverCert *shared.CertInfo, serverName string, targetAddress string, targetCert string, targetPassword string) error {
+	// Connect to the target cluster node.
+	args := &lxd.ConnectionArgs{
+		TLSServerCert: targetCert,
+		UserAgent:     version.UserAgent,
+	}
+
+	target, err := lxd.ConnectLXD(fmt.Sprintf("https://%s", targetAddress), args)
+	if err != nil {
+		return fmt.Errorf("Failed to connect to target cluster node %q: %w", targetAddress, err)
+	}
+
+	cert, err := generateTrustCertificate(serverCert, serverName)
+	if err != nil {
+		return fmt.Errorf("Failed generating trust certificate: %w", err)
+	}
+
+	post := api.CertificatesPost{
+		CertificatePut: cert.CertificatePut,
+		Password:       targetPassword,
+	}
+
+	err = target.CreateCertificate(post)
+	if err != nil && err.Error() != ErrCertificateExists.Error() {
```

Comment:
Does the lxd server response when certificate already exists use a specific error code as ideally we would detect it using api.StatusErrorCheck() here.

Reply:
Actually, looking into this more it looks like this error is long gone.

Reply:
Actually not, apparently I was grepping incorrectly :)

Reply:
OK, I've updated the error to use `409 Conflict` if a certificate already exists.